### PR TITLE
send task 5 for review

### PR DESCRIPTION
**Pull request by BeckBob**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -91,3 +91,43 @@ describe("GET /api/articles/:article_id", () => {
...
+    })
+    test("articles appear in descending order", () => {
+        return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({body}) => {
+            const articles = body.body
+            expect(articles[0].article_id).toBe(articles.length)
+        })
+    })
```
Comment (MKD01) [inline __tests__/endpoints.test.js]: I like the idea behind this test, but it is not a very robust test, I would recommend looking into the library jest-sorted for a nicer way to do this test assertion.

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -91,3 +91,43 @@ describe("GET /api/articles/:article_id", () => {
...
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                })
+            })
+        })
```
Comment (MKD01) [inline __tests__/endpoints.test.js]: This is a good test so far, but can give back a false pass when given an empty array, so to ensure this doesnt happen we need an assertion to ensure that the array is also not empty before entering the forEach.

File: __tests__/endpoints.test.js [hunk 3]
```diff
@@ -91,3 +91,43 @@ describe("GET /api/articles/:article_id", () => {
...
+    })
+    test("when given a bad query endpoint returns Bad Request error", () => {
+        return request(app)
+        .get("/api/articles?banana=banana")
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad Request")
+        })
+    })
```
Comment (MKD01) [inline __tests__/endpoints.test.js]: Im not sure what this test is checking for exactly, but at this point we shouldnt be expecting any user input so there is no valid error handling that needs to be added for this endpoint yet.

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -1,6 +1,15 @@
-const { getArticle, checkArticleExists } = require("../models/articles.model")
+const { getArticle, checkArticleExists, getAllArticlesObj } = require("../models/articles.model")
 
- 
+exports.getAllArticles = (req, res, next) => {
+    const query = req.query
+    var size = Object.keys(query).length;
```
Comment (MKD01) [inline controllers/articles.controller.js]: Please dont declare variables with var, using let/const is always going to be more favourable.

File: models/articles.model.js [hunk 5]
```diff
@@ -7,6 +7,33 @@ return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
...
+                    ON comments.article_id = articles.article_id
+                    GROUP BY articles.author,
+                    articles.title, 
+                    articles.article_id,
+                    articles.topic,
+                    articles.created_at,
+                    articles.votes,
+                    articles.article_img_url,
+                    comments.article_id
+                    ORDER BY articles.article_id DESC;`)
```
Comment (MKD01) [inline models/articles.model.js]: This is a good query but is a little larger than it needs to be, the GROUP BY does not need to contain so many column names for this to work.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Really good work so far, but there are some changes primarily in your testing that I would like to see before you merge this in, so please re-submit this pr once those changes have been added.

**Review by aloush-dev — APPROVED** [review]
Awesome job implementing feedback!
Merge when ready

**Pull request merged**